uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/src/leave-absences/absence-tab/components/absence-tab-custom-work-pattern-modal.component.js

```diff
@@ -107,12 +107,12 @@ define([
     }
 
     /**
-     * Loads the Job Contract Revision Change Reasons and indexes by `value`
+     * Loads the Leave and Absences Work Pattern Change Reasons and indexes by `value`
```

Comment:
I am not sure where does it index the loaded data by value.

Reply:
I assumed that this is referring to [the modal html file](https://github.com/civicrm/civihr/blob/staging/uk.co.compucorp.civicrm.hrleaveandabsences/views/absence-tab/components/absence-tab-custom-work-pattern-modal.html#L52) when updating the comment. Kindly let me know if that is wrong.

Reply:
This is a function that should not really care about the modal, it should not care about anything outside it. If it doesn't index and returns an array instead, the `and indexes by value` is misleading information. I think this is a broad practice in all programming languages that have functions.